extractors/aggregate_flows_from_google_data.py

```diff
@@ -0,0 +1,218 @@
+- data_processing_config.yaml, and
+- metadata.yaml
+metadata.yaml should include something like:
+```
+- doi_or_unique_name: mid-year-pop-est-18-tabs_Table 2.csv
+  filename: human/external/mid-year-pop-est-18-tabs_Table 2.csv
+- doi_or_unique_name: Global_Mobility_Report.csv
+  filename: human/external/Global_Mobility_Report.csv
+```
+This script can be run with
+```
+python aggregate_flows_from_google_data.py
+```
+The script generates .h5 files in generated_sns_products/movement_multiplier
+"""
+
+from pathlib import Path
+import pandas as pd
+import urllib.request
+from data_pipeline_api.data_processing_api import DataProcessingAPI
+from ftplib import FTP
+import zipfile
+
+
+config_filename =  Path(__file__).parent / "data_processing_config.yaml"
+uri = "data_processing_uri"
+git_sha = "data_processing_git_sha"
+data_path = Path(__file__).parent / "human/external"
+
+
+def download_pop_table():
+    """
+    Download the population data from an external source if it doesn't exists in data_path, using only the Area1 data and the first 3 columns
+    the LA code, name and population, removing the commas in the population numbers.
+    :return: A dataframe containing the local authority code, name and population
+             and the total polulation
+    """
+
+    population_table = "mid-year-pop-est-18-tabs_Table 2.csv"
+
+
+    # If the population table doesn't exist download it.
+    if not Path(data_path / population_table).exists():
+        print(f"Could not find {data_path}/{population_table}, downloading it")
+        url = "https://www.nrscotland.gov.uk/files//statistics/population-estimates/mid-18/mid-year-pop-est-18-tabs.zip"
+        zip_filename = "mid-year-pop-est-18-tabs.zip"
+        urllib.request.urlretrieve(
+            url, zip_filename
+        )
+
+        with zipfile.ZipFile(zip_filename, 'r') as zip_ref:
+            zip_ref.extractall(data_path, members=[population_table])
+
+        # clean up (i.e. remove) the downloaded datafile(s)
+        Path(zip_filename).unlink(missing_ok=False)
+
+    with DataProcessingAPI(config_filename, uri=uri, git_sha=git_sha) as api:
+        with api.read_external_object(population_table) as file:
+            dfPop = pd.read_csv(file, skiprows=5, nrows=32, usecols=[0, 1, 2])
+            dfPop.columns = ['la_code', 'la_name', 'population']
+            dfPop['population'] = dfPop['population'].str.replace(',', '').astype(int)
+
+    total_population =  dfPop['population'].sum()
+
+    return dfPop, total_population
+
+def download_lookup_table():
+    """
+    Downloads the mapping of iso_3166_2 codes to local authority from the SCRC database, 
+    if it doesn't exist, upload it
+    :return: A dataframe containing the full ISO code (GB-iso_3166_2) and the corresponding local authority code.
+    """
+    
+    #TODO - should I check if these files are in the pipeline and if not upload them before starting?
```

Comment:
For the moment I think the plan is to assume that it is available somewhere as an external object

Reply:
I've tidied up the TODOs and removed the in-script creation of the iso mapping file. If we ever need it we can get it from git.